### PR TITLE
Add remote Entropy driver for W91 based on IPC unit

### PR DESCRIPTION
**Pull request by serhiiSalamakha**
http://192.168.48.49:8080/browse/ZEPHYR-420

### CODE REVIEW

File: drivers/entropy/entropy_w91_trng.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+/* API implementation: get_entropy_isr */
+static int entropy_w91_trng_get_entropy_isr(const struct device *dev,
+					    uint8_t *buffer, uint16_t length,
+					    uint32_t flags)
+{
+	ARG_UNUSED(flags);
+
+	/* No specific handling in case of running from ISR, just call standard API */
+	return entropy_w91_trng_get_entropy(dev, buffer, length);
```
Comment (andriy-bilynskyy) [inline drivers/entropy/entropy_w91_trng.c]: It won't work: we are in ISR here.
There is no possibility to receive IPC response in ISR.
To overcome this I propouse to use standard c random here.
Init standard c random when we call it first time using 10MHz MTIMER value.
But at first plese measure timings of these function execition using ligic analyzer to ensure that we can spend this time in ISR.

File: drivers/entropy/entropy_w91_trng.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+					uint8_t *buffer, uint16_t length)
+{
+	struct entropy_w91_trng_get_entropy_resp trng_get_entropy_resp = {
+		.buffer	= buffer,
+	};
+
+	struct ipc_based_driver *ipc_data = &((struct entropy_w91_data *)dev->data)->ipc;
+	uint8_t inst = ((struct entropy_w91_config *)dev->config)->instance_id;
+
+	IPC_DISPATCHER_HOST_SEND_DATA(ipc_data, inst,
```
Comment (andriy-bilynskyy) [inline drivers/entropy/entropy_w91_trng.c]: As you can see buffer here can be big.
Cryptography ofthen requests 4K bufeer so it's no possible to send such amount of data in one communication cycle.
We should send data few times:
- send function returns amount of bytes we are able to send. if it more than our id (mesage valid) send next portion adding id before again. repeat till all data will be send.

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: drivers/entropy/entropy_w91_trng.c (hunk 2 above)
Comment (serhiiSalamakha) [reply drivers/entropy/entropy_w91_trng.c]: updated

File: drivers/entropy/entropy_w91_trng.c (hunk 1 above)
Comment (serhiiSalamakha) [reply drivers/entropy/entropy_w91_trng.c]: This approach will create delays at large values (if the user requests 4K bytes), so it is better to specify that getting entropy is not supported in interrupts.

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**